core/src/rpc/format/mod.rs

```diff
@@ -0,0 +1,82 @@
+mod bincode;
+pub mod cbor;
+mod json;
+pub mod msgpack;
+mod revision;
+
+use ::revision::Revisioned;
+use serde::Serialize;
+
+use super::{request::Request, RpcError};
+use crate::sql::Value;
+
+pub const PROTOCOLS: [&str; 5] = [
```

Comment:
Im not sure about how this is done. It means the indexes further down (lines 39+) are hard-coded ordering.

Maybe this could be a `Lazy<BtreeMap<&'static str, Format>>` instead?

Reply:
PROTOCOLS is required to implement Iterator, where Item: Into<Cow<'static, &str>>, it could be a Lazy<BTreeSet<&static str>>, but I don't see what the benefit would be

Reply:
I see @phughk's point, this could be improved in a future PR. Ideally we wouldn't want the Format enums to mismatch the strings.